### PR TITLE
Dynamically symlink all config/dotfiles to home directory

### PR DESCRIPTION
**Pull request by christippett**
All files located in the config/ folder will be symlinked to a user's home directory.

Features:
- Respects file path hierarchies (e.g. ~/.ssh/config)
- Removes existing file symlink before re-applying
- Backups/renames existing config files (appends `.[YYYYMMDD].bak` to filename)

### CODE REVIEW

**Comment by christippett** [comment]
Address #23

**Comment by christippett** [comment]
I ended up refactoring this completely ha

File: scripts/setup-shell.bash [hunk 1]
```diff
@@ -58,30 +58,18 @@ git clone https://github.com/zsh-users/zsh-syntax-highlighting.git ~/.oh-my-zsh/
...
 
-# symlink aliases
-log_info "ℹ️  Symlinking aliases"
-ln -sv ~/projects/dotfiles/config/.aliases ~/.aliases
-
 # starship theme
-
 log_info "🚀  Installing Starship theme"
-mkdir -p ~/.config
-ln -sv ~/projects/dotfiles/config/starship.toml ~/.config/starship.toml
```
Comment (jthegedus) [inline scripts/setup-shell.bash:84]: This is the default dir and so an ENV_VAR would need to be set with the new value of `~/starhsip.toml` - https://starship.rs/config/#configuration

Comment (jthegedus) [reply scripts/setup-shell.bash:84]: wait, I see what you've done. Clever

**Pull request merged**